src/django_components/extension.py

```diff
@@ -29,7 +30,7 @@
 
 # Mark a class as an extension hook context so we can place these in
 # a separate documentation section
-def mark_extension_hook_api(cls: Type[Any]) -> Type[Any]:
+def mark_extension_hook_api(cls: TClass) -> TClass:
```

Comment:
Unrelated to the PR, but I noticed that some type hints didn't work. And I realized it was because of these decorators that I made for taging functions and classes for documentation.

By making the decorator to take and return generics, the decorator no longer changes the type of the classes / functions, and everything works well again.